### PR TITLE
Stream sync test kernel

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Adds test to stream sync changes for kernel

### CODE REVIEW

File: test/gpu/stream_sync.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+    migraphx::gpu::kernel k1{binaries.front(), "compare"};
+
+    auto input =
+        migraphx::fill_argument({migraphx::shape::float_type, {128}}, STREAM_SYNC_TEST_VAL);
+
+    auto ginput = migraphx::gpu::to_gpu(input);
+
+    hipStream_t stream;
+    auto status = hipStreamCreate(&stream);
```
Comment (pfultz2) [inline test/gpu/stream_sync.cpp:81]: I would make a function to return the stream as a managed pointer.

File: test/gpu/stream_sync.cpp [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    if(status != hipSuccess)
+    {
+        MIGRAPHX_THROW("Failed to get stream");
+    }
+
+    k1.launch(stream, input.get_shape().elements(), 1024)(ginput.cast<std::float_t>());
```
Comment (pfultz2) [inline test/gpu/stream_sync.cpp:87]: Just use `float` instead `std::float_t`. I didnt even know that was a valid type.

File: test/gpu/stream_sync.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    migraphx::program p;
+    auto* mm      = p.get_main_module();
+    auto x        = mm->add_parameter("x", io_shape);
+    auto y        = mm->add_literal({io_shape, {STREAM_SYNC_TEST_VAL + 100}});
+    auto test_val = mm->add_literal({io_shape, {STREAM_SYNC_TEST_VAL}});
+
+    auto mult_out = mm->add_instruction(migraphx::make_op("mul"), x, y);
+    auto add_out  = mm->add_instruction(migraphx::make_op("add"), mult_out, test_val);
+    mm->insert_parameter(add_out, "output", io_shape);
```
Comment (pfultz2) [inline test/gpu/stream_sync.cpp:137]: Dont insert the output parameter. That should be added when compiling.

File: test/gpu/stream_sync.cpp [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+    migraphx::compile_options options;
+    p.compile(migraphx::gpu::target{}, options);
```
Comment (pfultz2) [inline test/gpu/stream_sync.cpp:140]: This can just be `p.compile(migraphx::gpu::target{})` since its just using the default options.

Comment (TedThemistokleous) [reply test/gpu/stream_sync.cpp]: Done

File: test/gpu/stream_sync.cpp (hunk 3 above)
Comment (TedThemistokleous) [reply test/gpu/stream_sync.cpp]: Done

File: test/gpu/stream_sync.cpp [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    }
+
+    migraphx::program p;
+    auto* mm      = p.get_main_module();
+    auto x        = mm->add_parameter("x", io_shape);
+    auto y        = mm->add_literal({io_shape, {STREAM_SYNC_TEST_VAL + 100}});
+    auto test_val = mm->add_literal({io_shape, {STREAM_SYNC_TEST_VAL}});
+
+    auto mult_out = mm->add_instruction(migraphx::make_op("mul"), x, y);
+    auto add_out  = mm->add_instruction(migraphx::make_op("add"), mult_out, test_val);
```
[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply test/gpu/stream_sync.cpp]: Replaced to dot, and adjusted sizes

File: test/gpu/stream_sync.cpp (hunk 1 above)
Comment (TedThemistokleous) [reply test/gpu/stream_sync.cpp:73]: Done. Just reused what we did with hip_event_ptr but for streams.

**Review by umangyadav — APPROVED** [review]

**Pull request merged**